### PR TITLE
Refactor `backend.js` #5 - Simplify the main logic

### PR DESCRIPTION
**Pull request by nemanjaglumac**
### Status
PENDING CI && PENDING REVIEW

### What does this PR accomplish?
- This is the fifth step/task in https://github.com/metabase/metabase/issues/24023
- Greatly simplifies the previous logic for `BackendResource` whose only job was to spin up (temporary) backend for Cypress

> **Note**
> I tried to keep changes logically contained to a single commit. That's probably the easiest way to review this PR, rather than trying to look at the final diff.

As before, everything should still work the same as before when you spin Cypress up (both locally and in the CI).

### Questions:
What do you think about simplifying this even further and dealing with three separate functions that we can export from `cypress-runner-backend.js`?

Something like:
```js
// cypress-runner-backend.js

module.exports = { createServer, startBackend, stopBackend }
```

and then using them in `run_cypress_tests.js` like so:
```js
const server = createServer(/*we can pass options for the server directly here*/)

startBakend(server, /* also possible to pass even more additional options, or flags for the backend if needed */)

// and in the end
stopBackend(server)
```

### Additional info:
- If we decide to go down this route, then this PR would replace https://github.com/metabase/metabase/pull/24077 created by @ariya as an alternative approach
- I didn't compare the times explicitly, but it seems like backend is now spawning MUCH faster

### CODE REVIEW

**Comment by gusaiani** [comment]
@nemanjaglumac 

> module.exports = { createServer, startBackend, stopBackend }

Yes yes yes 👍

**Review by gusaiani — APPROVED** [review]
Looking better and better.

File: frontend/test/__runner__/cypress-runner-backend.js [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+const fs = require("fs");
+const os = require("os");
+const path = require("path");
+const { spawn } = require("child_process");
+
+const fetch = require("isomorphic-fetch");
+
+const CypressBackend = e2eBackend();
+
+function e2eBackend() {
```
Comment (npfitz) [inline frontend/test/__runner__/cypress-runner-backend.js:12]: I think we can remove a layer here and just have `CypressBackend` set to what `e2eBackend()`, without actually calling `e2eBackend`
```suggestion
const CypressBackend = {
 createServer() {...},
 start(server) {...},
 ...
}
```

Comment (npfitz) [reply frontend/test/__runner__/cypress-runner-backend.js:12]: Haha, @gusaiani beat me to it

Comment (nemanjaglumac) [reply frontend/test/__runner__/cypress-runner-backend.js]: @npfitz done! Thank you for the suggestion.

**Pull request merged**